bumble/gatt_client.py

```diff
@@ -78,16 +92,16 @@
 # Proxies
 # -----------------------------------------------------------------------------
 class AttributeProxy(EventEmitter):
-    client: Client
-
-    def __init__(self, client, handle, end_group_handle, attribute_type):
+    def __init__(
+        self, client: Client, handle: int, end_group_handle: int, attribute_type: UUID
+    ) -> None:
         EventEmitter.__init__(self)
         self.client = client
         self.handle = handle
         self.end_group_handle = end_group_handle
         self.type = attribute_type
 
-    async def read_value(self, no_long_read=False):
+    async def read_value(self, no_long_read: bool = False) -> bytes:
```

Comment:
Because of the polymorphic nature of `decode`, the return type here should really be `Any`, because `AttributeProxy` is a generic superclass, characteristic adapters may decode byte arrays into any type (profile dependent). It's not easy to see it, because that happens though the CharacteristicAdapter class that does some of the patching at runtime (which is always confusing for the type checker).